src/main/java/com/google/devtools/build/lib/util/DependencySet.java

```diff
@@ -72,10 +73,18 @@ public void setOutputFileName(String outputFileName) {
   /**
    * Constructs a new empty DependencySet instance.
    */
-  public DependencySet(Path root) {
+  public DependencySet(OS os, Path root) {
```

Comment:
(Argh the ordering of methods in this file).

Every DependencySet seems to be created from an Action (either NinjaAction or CppCompileAction), so we can get the execution platform from that.